drivers/console/uart_console.c

```diff
@@ -86,11 +87,25 @@ static int console_out(int c)
 
 #endif  /* CONFIG_UART_CONSOLE_DEBUG_SERVER_HOOKS */
 
+	if (pm_device_runtime_is_enabled(uart_console_dev)) {
+		if (pm_device_runtime_get(uart_console_dev) < 0) {
```

Comment:
I know this is unrelated but having to check `pm_device_runtime_is_enabled` before calling `pm_device_runtime_*` is ugly to my eyes. The `pm_device_runtime_*` should simply return some error code if PM runtime is not enabled.

Reply:
If the device_runtime is not enabled, pm_device_runtime_get / put will just return 0 since the device is active and can be used. For pm_device_runtime_get that is pretty reasonable, for pm_device_runtime_put some can argue that it should return something different. Though, checking for pm_device_runtime_is_enabled() seems not needed. We could just ignore it and call directly pm_device_runtime_get, if it fails it will just return `c` if not the function will ignore this block as if pm_device_runtime_is_enabled() returned false.